activiti-core/activiti-engine/src/main/java/org/activiti/engine/impl/repository/DeploymentBuilderImpl.java

```diff
@@ -97,6 +98,19 @@ public boolean hasProjectManifestSet(){
       return this.projectManifest != null;
   }
 
+  public DeploymentBuilder setEnforcedAppVersion(Integer enforcedAppVersion){
+      this.enforcedAppVersion = enforcedAppVersion;
+      return this;
+  }
+
+  public Integer getEnforcedAppVersion(){
+      return this.enforcedAppVersion;
+  }
+
+  public boolean hasEnforcedAppVersion(){
+      return this.enforcedAppVersion != null;
```

Comment:
Why is in here a different type of check? Isn't this supposed to behave the same way? We should be using different names for different concepts and same names only to the same concepts

Reply:
this `enforcedAppVersion` check is done after the  enforcedAppVersion loading from the `applicationUpgradeContextService`
they hold the same value, but they are different fields belonging to different logic layers.
Hence, their default status is different (0 for the `applicationUpgradeContextService`, and null for this one in the DeploymentBuilder